sympy/utilities/tests/test_lambdify.py

```diff
@@ -414,44 +414,66 @@ def test_cmath_atanh():
-                        modules=[custom_mappings, "cmath", "math"])
-    assert func_exp(hpi + 1j * hpi) == 0j
+                        modules=["sympy", "cmath", "math"])
+    assert abs(func_exp(hpi + 1j * hpi)) < 4e-16
 
-    # Complex Cosine Identity: cos(z) = cos(Re(z)) * cosh(Im(z)) - i sin(Re(z)) * sinh(Im(z))
+    # Complex Cosine Identity: cos(z) = cos(Re(z)) * cosh(Im(z)) - i*sin(Re(z)) * sinh(Im(z))
     func_cos = lambdify([z], cos(z) - (cos(re(z)) * cosh(im(z)) - I * sin(re(z)) * sinh(im(z))),
-                        modules=[custom_mappings, "cmath", "math"])
-    assert func_cos(hpi + 1j * hpi) == 0j
+                        modules=["sympy", "cmath", "math"])
+    assert abs(func_cos(hpi + 1j * hpi)) < 4e-16
 
-    # Complex Sine Identity: sin(z) = sin(Re(z)) * cosh(Im(z)) + i cos(Re(z)) * sinh(Im(z))
+    # Complex Sine Identity: sin(z) = sin(Re(z)) * cosh(Im(z)) + i*cos(Re(z)) * sinh(Im(z))
     func_sin = lambdify([z], sin(z) - (sin(re(z)) * cosh(im(z)) + I * cos(re(z)) * sinh(im(z))),
-                        modules=[custom_mappings, "cmath", "math"])
-    assert func_sin(hpi + 1j * hpi) == 0j
+                        modules=["sympy", "cmath", "math"])
+    assert abs(func_sin(hpi + 1j * hpi)) < 4e-16
 
-    # Complex Hyperbolic Cosine Identity: cosh(z) = cosh(Re(z)) * cos(Im(z)) + i sinh(Re(z)) * sin(Im(z))
+    # Complex Hyperbolic Cosine Identity: cosh(z) = cosh(Re(z)) * cos(Im(z)) + i*sinh(Re(z)) * sin(Im(z))
     func_cosh = lambdify([z], cosh(z) - (cosh(re(z)) * cos(im(z)) + I * sinh(re(z)) * sin(im(z))),
-                         modules=[custom_mappings, "cmath", "math"])
-    assert func_cosh(hpi + 1j * hpi) == 0j
+                         modules=["sympy", "cmath", "math"])
+    assert abs(func_cosh(hpi + 1j * hpi)) < 4e-16
 
-    # Complex Hyperbolic Sine Identity: sinh(z) = sinh(Re(z)) * cos(Im(z)) + i cosh(Re(z)) * sin(Im(z))
+    # Complex Hyperbolic Sine Identity: sinh(z) = sinh(Re(z)) * cos(Im(z)) + i*cosh(Re(z)) * sin(Im(z))
     func_sinh = lambdify([z], sinh(z) - (sinh(re(z)) * cos(im(z)) + I * cosh(re(z)) * sin(im(z))),
-                         modules=[custom_mappings, "cmath", "math"])
-    assert func_sinh(hpi + 1j * hpi) == 0j
+                         modules=["sympy", "cmath", "math"])
+    assert abs(func_sinh(hpi + 1j * hpi)) < 4e-16
 
+    # cosh(z) = (e^z + e^(-z)) / 2
     func_cosh = lambdify([z], cosh(z) - (exp(z) + exp(-z)) / 2, modules=["cmath", "math"])
-    assert func_cosh(hpi) == 0j
+    assert abs(func_cosh(hpi)) < 4e-16
+
+    # Additional expressions testing log and exp with real and imaginary parts
+    expr1 = log(re(z)) + log(im(z)) - log(re(z) * im(z))
+    expr2 = exp(re(z)) * exp(im(z) * I) - exp(z)
+    expr3 = log(exp(re(z))) - re(z)
+    expr4 = exp(log(re(z))) - re(z) 
+    expr5 = log(exp(re(z) + im(z))) - (re(z) + im(z))
+    expr6 = exp(log(re(z) + im(z))) - (re(z) + im(z))
+    func1 = lambdify([z], expr1, modules=["sympy", "cmath", "math"])
```

Comment:
we should not add "sympy" to the list of printers, then it's no longer testing out CMathPrinter, perhaps we should name this test something with "cmath" in its name?